doc/developer/design/20230512_mir_cost_model.md

```diff
@@ -151,3 +171,13 @@ How should we communicate source features to the cost model?
 [future-work]: #future-work
 
 When we have a candidate cost model, we will want to identify opportunities to use the cost model in the optimizer and pick a good first place to use it—most likely join planning. Actually using the cost model in the optimizer will require more validation and testing—at a minimum on LDBC, but ideally on client queries—to ensure that we don’t regress performance.
+
+We've identified the following early candidate clients for the cost model:
```

Comment:
I think future readers might gain clarity for each application we list which cost dimensions will be required. My current understanding:

1. **Join ordering.** requires `M` (only for `MirRelationExpr::Join`), `C` over `MirRelationExpr`, and `P` over `MirScalarExpr`.
2. **CTE filter pushdown.** requires `C` over `MirRelationExpr` and `P` over `MirScalarExpr`.
3. **Reduction pushdown.** requires `C` over `MirRelationExpr` and `P` over `MirScalarExpr`.
4. **Late materialization.** requires `M` over `MirRelationExpr` and `P` over `MirScalarExpr` (and the former might have to be based on a throw-away lowering as indicated above).